### PR TITLE
Add strongly connected components with Kosaraju solver

### PR DESCRIPTION
**Pull request by surendrathapa18**
Implemented Strongly Connected Components using Kosaraju’s Algorithm.

Team Members:
- Surendra Thapa
- Rohan Shrestha

Includes:
- P_STRONGLYCONNECTEDCOMPONENTS problem
- KosarajuSolver
- SCCVerifier
- Unit tests (multiple cases)

All tests passed (dotnet test).

### CODE REVIEW

File: redux-tests/Problems/P_STRONGLYCONNECTEDCOMPONENTS/P_STRONGLYCONNECTEDCOMPONENTS_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            var problem = new P_STRONGLYCONNECTEDCOMPONENTS(input);
+            var solver = new KosarajuSolver();
+
+            string result = solver.solve(problem);
+
+            Assert.Contains("{1,2,3}", result);
+        }
+        [Fact]
+public void Kosaraju_Single_Node_Graph()
+{
```
Comment (wrigjl) [inline redux-tests/Problems/P_STRONGLYCONNECTEDCOMPONENTS/P_STRONGLYCONNECTEDCOMPONENTS_Tests.cs]: the indention level for this function is wrong

**Review by wrigjl — DISMISSED** [review]

**Comment by Andrija-Sevaljevic** [comment]
Another note: Please put tests in appropriate folder.

File: redux-tests/Problems/P_STRONGLYCONNECTEDCOMPONENTS/P_STRONGLYCONNECTEDCOMPONENTS_Tests.cs (hunk 1 above)
Comment (surendrathapa18) [reply redux-tests/Problems/P_STRONGLYCONNECTEDCOMPONENTS/P_STRONGLYCONNECTEDCOMPONENTS_Tests.cs]: Fixed the indentation issue for the SCC test. Please review again. Thanks!

**Comment by surendrathapa18** [comment]
Indentation issue fixed and tests are in correct folder now. Please review again. Thanks!

**Comment by Andrija-Sevaljevic** [comment]
Documentation - Proper sourcing, formal problem definition etc. needs additional work. Look at CLIQUE_Class.cs for best practices.
Implementation - The verifier depends on the exact solution, but the set ordering is not unique. Bug in verifier that needs to be addressed urgently.
Testing - Missing verifier tests, edge case depth, general lack of comprehensive tests.

Work on these three things and it will be merge ready. I have also made a commit so it shows up on Redux (renaming of classes and files, just pull, no need to modify further for now).

**Comment by Andrija-Sevaljevic** [comment]
LGTM! You may improve verifier, solver and problem source to an actual published paper, as well as proper linkage.
